### ISSUE TITLE
`Too many open files` error from UDF server

### ISSUE THREAD

**Issue by lmatz** [risingwavelabs/risingwave#9074]
### Describe the bug

```
E0410 04:40:33.165965010  321185 tcp_server_posix.cc:205]    Failed accept4: Too many open files
```

Not sure if it has anything to do with the upper limit of the number of file descriptors.

Probably there is leak of connections.

### Additional context

Reported by one design partner.

**Comment by wangrunji0408** [comment]
Seems to no longer happen after #9240.

**Comment by lmatz** [comment]
The error seems to still exist, in our cluster and a user's cluster.

**Comment by wangrunji0408** [comment]
It should have been fixed by #9916.
